src/helpers/transaction.ts

```diff
@@ -135,32 +159,59 @@ export const makeFinishedTxReceipt = (
   confirmations
 });
 
-const decodeTransaction = (signedTx: BytesLike) => {
-  const decodedTransaction = parseTransaction(signedTx);
-  const gasLimit = bigNumGasLimitToViewable(decodedTransaction.gasLimit.toString());
-  const gasPriceGwei = bigNumGasPriceToViewableGwei(decodedTransaction.gasPrice.toString());
-  const amountToSendEther = bigNumValueToViewableEther(decodedTransaction.value.toString());
-
-  return {
-    to: decodedTransaction.to,
-    from: decodedTransaction.from,
-    value: amountToSendEther.toString(),
-    gasLimit: gasLimit.toString(),
-    gasPrice: gasPriceGwei.toString(),
-    nonce: decodedTransaction.nonce,
-    data: decodedTransaction.data,
-    chainId: decodedTransaction.chainId
-  };
-};
+const getGasPriceFromTx = (tx: {
+  type?: number | null;
+  maxFeePerGas?: BigNumber | string;
+  maxPriorityFeePerGas?: BigNumber | string;
+  gasPrice?: BigNumber | string;
+}) =>
+  // Possibly revisit this when more tx types are available
+  tx.type && tx.type == 2
+    ? {
+        maxFeePerGas: hexlify(tx.maxFeePerGas!) as ITxGasPrice,
+        maxPriorityFeePerGas: hexlify(tx.maxPriorityFeePerGas!) as ITxGasPrice,
+        type: tx.type
+      }
+    : { gasPrice: hexlify(tx.gasPrice!) as ITxGasPrice };
```

Comment:
Could we use a switch for that so we can easily plug another type in the future

Reply:
Since it is only for gas, I don't expect there to be any changes to that in the future. This would be fine for now, eventually we could expand it to just default to EIP 1559 and fallback to the old gas price. Because newer transaction types probably will use EIP 1559 gas.

Reply:
Okay good then :)